src/main/java/in/projecteka/consentmanager/user/LockedUserService.java

```diff
@@ -25,12 +25,16 @@
         return lockedUsersRepository.deleteUser(cmId);
     }
 
-    public Mono<Void> createOrUpdateLockedUser(String cmId) {
+    public Mono<Integer> createOrUpdateLockedUser(String cmId) {
         return lockedUsersRepository.getLockedUserFor(cmId)
-                .flatMap(lockedUser ->
-                        isBeforeMinutes(lockedUser.getDateCreated(), lockedServiceProperties.getCoolOfPeriod())
-                                ? Mono.empty() : removeLockedUser(cmId))
-                .then(lockedUsersRepository.upsert(cmId));
+                .flatMap(lockedUser -> {
+                    var remainingTries = lockedServiceProperties.getMaximumInvalidAttempts() - lockedUser.getInvalidAttempts();
+                    return isBeforeMinutes(lockedUser.getDateCreated(), lockedServiceProperties.getCoolOfPeriod())
+                            ? Mono.just(remainingTries - 1)
+                            : removeLockedUser(cmId).then(Mono.just(lockedServiceProperties.getMaximumInvalidAttempts() - 1));
+                })
+                .flatMap(remainingTries -> lockedUsersRepository.upsert(cmId).thenReturn(remainingTries))
+                .switchIfEmpty(Mono.defer(() -> lockedUsersRepository.upsert(cmId).then(Mono.just(lockedServiceProperties.getMaximumInvalidAttempts() - 1))));
```

Comment:
We can use `thenReturn` here. `then(Mono.just(lockedServiceProperties.getMaximumInvalidAttempts() - 1))`

Reply:
yeah off course. will change it